### PR TITLE
Symfonize agent index

### PR DESCRIPTION
**Pull request by sillydebs**

### CODE REVIEW

**Comment by sillydebs** [comment]
No errors detected with debug mode

**Review by alexarn — CHANGES_REQUESTED** [review]

**Comment by alexarn** [comment]
Html entities are removed in the atomic update but not in db_data. So with a new installation, i get them in the list of status and services

File: templates/agents/agent_table_edit.html.twig [hunk 1]
```diff
@@ -146,9 +146,9 @@
       <td>
         {% for s in sites_select %}
           {% if s.checked %}
-            <input type='checkbox' name='sites[]' value='{{ s.id }}' checked="checked" >{{ s.name | raw }}<br/>
+            <input type='checkbox' name='sites[]' value='{{ s.id }}' checked="checked" >{{ s.name }}<br/>
           {% else %}
-            <input type='checkbox' name='sites[]' value='{{ s.id }}'>{{ s.name | raw }}<br/>
+            <input type='checkbox' name='sites[]' value='{{ s.id }}'>{{ s.name }}<br/>
```
[the remark this answers is not in the document]
Comment (sillydebs) [reply templates/agents/agent_table_edit.html.twig:151]: There is : the rendering is okay with specific characters without the filter.

[the remark this answers is not in the document]
Comment (sillydebs) [reply templates/agents/agent_table_edit.html.twig:151]: No, but I found out this filter is required in other templates. Thank you !

**Comment by alexarn** [comment]
Issue found with batch editing feature:

If i change "statut" and/or "service", i get the html encoded version.

File: public/include/db.php [hunk 2]
```diff
@@ -98,7 +99,11 @@ public function query($requete)
                     if (isset($isCryptedPassword) and $isCryptedPassword===true) {
                         $result[$key]=filter_var($value, FILTER_UNSAFE_RAW);
                     } else {
-                        $result[$key]=filter_var($value, FILTER_SANITIZE_STRING);
+                        if ($this->sanitize_string) {
+                            $result[$key] = filter_var($value, FILTER_SANITIZE_STRING);
+                        } else {
+                            $result[$key] = filter_var($value, FILTER_UNSAFE_RAW);
+                        }
```
Comment (alexarn) [inline public/include/db.php:106]: Please remove these changes. There are already on master

File: src/Controller/AgentController.php [hunk 3]
```diff
@@ -599,17 +754,17 @@ public function save(Request $request)
...
+        $nom = trim($params['nom']);
         $postes = $params['postes'];
         $prenom = trim($params['prenom']);
         $recup = isset($params['recup']) ? trim($params['recup']) : null;
-        $service = htmlentities($params['service'], ENT_QUOTES|ENT_IGNORE, 'UTF-8', false);
+        $service = $params['service'];
         $sites = array_key_exists("sites", $params) ? $params['sites'] : null;
-        $statut = htmlentities($params['statut'], ENT_QUOTES|ENT_IGNORE, 'UTF-8', false);
-        $temps=array_key_exists("temps", $params) ? $params['temps'] : null;
+        $statut = $params['statut'];
```
Comment (alexarn) [inline src/Controller/AgentController.php:763]: Restore html entities. Html entites will be removed in an other pull request

File: templates/agents/index.html.twig [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+              <a href='javascript:popup("personnel/suppression.php&amp;id={{ a.id }}",450,240);'>
+                <span class='pl-icon pl-icon-drop' title='Supprimer'></span>
+              </a>
+            {% endif %}
+          </td>
+          <td>{{ a.name }}</td>
+          <td>{{ a.surname }} </td>
+          <td>{{ a.hours }}</td>
+          <td>{{ a.status }}</td>
+          <td>{{ a.service }}</td>
```
Comment (alexarn) [inline templates/agents/index.html.twig:75]: Add raw filter on service and status for a good encoding

**Review by alexarn — CHANGES_REQUESTED** [review]
I think all the "sanitize_string = false" are useless now. Please check if they are needed

File: public/include/db.php [hunk 5]
```diff
@@ -98,7 +99,7 @@ public function query($requete)
                     if (isset($isCryptedPassword) and $isCryptedPassword===true) {
                         $result[$key]=filter_var($value, FILTER_UNSAFE_RAW);
                     } else {
-                        $result[$key]=filter_var($value, FILTER_SANITIZE_STRING);
+                        $result[$key] = filter_var($value, FILTER_SANITIZE_STRING);
                     }
                     $isCryptedPassword=($key=="type" and $value=="password")?true:false;
                 }
```
Comment (alexarn) [inline public/include/db.php:105]: Can you discard changes on this file please ?

File: public/personnel/ajax.update.php [hunk 6]
```diff
@@ -42,12 +42,11 @@
 $contrat = filter_input(INPUT_POST, 'contrat', FILTER_SANITIZE_STRING);
 $heures_hebdo = filter_input(INPUT_POST, 'heures_hebdo', FILTER_SANITIZE_STRING);
 $heures_travail = filter_input(INPUT_POST, 'heures_travail', FILTER_SANITIZE_STRING);
-$service = filter_input(INPUT_POST, 'service', FILTER_SANITIZE_STRING);
-$statut = filter_input(INPUT_POST, 'statut', FILTER_SANITIZE_STRING);
+$service = filter_input(INPUT_POST, 'service');
+$statut = filter_input(INPUT_POST, 'statut');
```
Comment (alexarn) [inline public/personnel/ajax.update.php:46]: Remove these changes please

File: public/personnel/ajax.update.php [hunk 7]
```diff
@@ -42,12 +42,11 @@
...
 
 $contrat = htmlentities($contrat, ENT_QUOTES|ENT_IGNORE, 'UTF-8', false);
-$service = htmlentities($service, ENT_QUOTES|ENT_IGNORE, 'UTF-8', false);
-$statut = htmlentities($statut, ENT_QUOTES|ENT_IGNORE, 'UTF-8', false);
```
Comment (alexarn) [inline public/personnel/ajax.update.php:50]: Restore htmlentities please

File: public/personnel/valid.php [hunk 8]
```diff
@@ -302,4 +302,4 @@ function login($nom, $prenom)
         $db->select("personnel", null, "login='$login'");
     }
     return $login;
-}
+}
```
Comment (alexarn) [inline public/personnel/valid.php:305]: Does this file already in use

File: public/setup/atomicupdate/updatepath.php [hunk 9]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+$sql[]="UPDATE `{$dbprefix}acces`SET `page` = '' WHERE `page`='personnel/index.php';";
```
Comment (alexarn) [inline public/setup/atomicupdate/updatepath.php:3]: Juste remove the page from this entry

**Review by alexarn — CHANGES_REQUESTED** [review]

File: public/personnel/valid.php (hunk 8 above)
Comment (sillydebs) [reply public/personnel/valid.php:305]: I don't know, but I cannot restore it.

File: public/setup/atomicupdate/updatepath.php (hunk 9 above)
Comment (sillydebs) [reply public/setup/atomicupdate/updatepath.php:3]: I do not understand your point, could you please explain it to me ?

File: src/Controller/AgentController.php [hunk 10]
```diff
@@ -599,17 +748,17 @@ public function save(Request $request)
         $mailsResponsables = trim(str_replace(array("\n", " "), null, $params['mailsResponsables']));
         $matricule = trim($params['matricule']);
         $url_ics = isset($params['url_ics']) ? trim($params['url_ics']) : null;
-        $nom=trim($params['nom']);
+        $nom = trim($params['nom']);
         $postes = $params['postes'];
         $prenom = trim($params['prenom']);
         $recup = isset($params['recup']) ? trim($params['recup']) : null;
-        $service = htmlentities($params['service'], ENT_QUOTES|ENT_IGNORE, 'UTF-8', false);
+        $service = $params['service'];
```
[the remark this answers is not in the document]
Comment (alexarn) [reply src/Controller/AgentController.php:755]: Please restore html entities !

**Review by alexarn — CHANGES_REQUESTED** [review]

File: src/Controller/AgentController.php (hunk 10 above)
[the remark this answers is not in the document]
Comment (sillydebs) [reply src/Controller/AgentController.php]: Thank you for the reminder. It has been corrected.

File: public/personnel/valid.php (hunk 8 above)
Comment (alexarn) [inline public/personnel/valid.php:305]: Please, can you remove this file ?

File: public/setup/atomicupdate/updatepath.php [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+
+$sql[]="UPDATE `{$dbprefix}menu` SET `url` = '/agent' WHERE `url`='personnel/index.php';";
```
Comment (alexarn) [inline public/setup/atomicupdate/updatepath.php:5]: Can you remove personnel/valid.php route please ?

**Review by alexarn — CHANGES_REQUESTED** [review]
1) Simple are displayed as htmlentities in skills batch edition

2) Service and status are wrong encoded in batch edition

**Comment by alexarn** [comment]
Merged on master. Thx Sarah !

**Pull request closed without merging**